linera-base/src/data_types.rs

```diff
@@ -277,6 +277,14 @@ pub struct Resources {
     pub bytes_to_read: u32,
     /// A number of bytes to write.
     pub bytes_to_write: u32,
+    /// A number of blobs to read.
+    pub blobs_to_read: u32,
+    /// A number of blobs to publish.
+    pub blobs_to_publish: u32,
```

Comment:
I'm not sure these two should have seaprate config values. I think the blobs reading/writing (publishing) should be limited by the blob bytes limits. I.e. it shouldn't matter if we're reading 1 blob twice the size of two smaller blobs half the size.

Reply:
I imagine that the cost of reading a 10-byte blob is almost the same as reading a 0-byte blob: Both require a roundtrip to the database, and very little payload. Also, reading a 0-byte blob shouldn't be free, since it does incur a cost. And we need to charge something for when a contract asserts the existence of a blob.

Reply:
There should be a constant cost for reaching to a db and then the per-byte cost.

Reply:
We are.
This struct here is only used for _grants_, where an application wants to pay for the execution of a message elsewhere, up to a limit. That limit is specified in resources.

Reply:
👍 

I'd make all limits gas-based.